snapcraft/parts/plugins/kernel.py

```diff
@@ -0,0 +1,1683 @@
+      (boolean; default: False)
+      use this flag to build in zfs support through extra ko modules
+
+    - kernel-enable-perf
+      (boolean; default: False)
+      use this flag to build the perf binary
+
+    - kernel-initrd-modules:
+      (array of string)
+      list of modules to include in initrd; note that kernel snaps do not
+      provide the core boot logic which comes from snappy Ubuntu Core
+      OS snap. Include all modules you need for mounting rootfs here.
+
+    - kernel-initrd-configured-modules:
+      (array of string)
+      list of modules to be added to the initrd
+      /lib/modules-load.d/ubuntu-core-initramfs.conf config
+      to be automatically loaded.
+      Configured modules are automatically added to kernel-initrd-modules.
+      If module in question is not supported by the kernel, it's automatically
+      removed.
+
+    - kernel-initrd-firmware:
+      (array of string)
+      list of firmware files to be included in the initrd; these need to be
+      relative paths to stage directory.
+      <stage/part install dir>/firmware/* -> initrd:/lib/firmware/*
+
+    - kernel-initrd-compression:
+      (string; default: as defined in ubuntu-core-initrd(zstd)
+      initrd compression to use; the only supported values now are
+      'lz4', 'xz', 'gz', 'zstd'.
+
+    - kernel-initrd-compression-options:
+      Optional list of parameters to be passed to compressor used for initrd
+      (array of string): defaults are
+        gz:  -7
+        lz4: -9 -l
+        xz:  -7
+        zstd: -1 -T0
+
+    - kernel-initrd-overlay
+      Optional overlay to be applied to built initrd.
+      This option is designed to provide easy way to apply initrd overlay for
+      cases modifies initrd scripts for pre uc20 initrds.
+      Value is relative path, in stage directory. and related part needs to be
+      built before initrd part. During build it will be expanded to
+      ${CRAFT_STAGE}/{initrd-overlay}
+      Default: none
+
+    - kernel-initrd-addons
+      (array of string)
+      Optional list of files to be added to the initrd.
+      Function is similar to kernel-initrd-overlay, only it works on per file
+      selection without a need to have overlay in dedicated directory.
+      This option is designed to provide easy way to add additional content
+      to initrd for cases like full disk encryption support, when device
+      specific hook needs to be added to the initrd.
+      Values are relative path from stage directory, so related part(s)
+      need to be built before kernel part.
+      During build it will be expanded to
+      ${CRAFT_STAGE}/{initrd-addon}
+      Default: none
+
+    - kernel-add-ppa
+      (boolean; default: True)
+      controls if the snappy-dev PPA should be added to the system
+
+This plugin support cross compilation, for which plugin expects
+the build-environment is comfigured accordingly and has foreign architectures
+setup accordingly.
+"""
+
+import logging
```

Comment:
Snapcraft components should use `craft_cli` instead of `logging`, you can replace `logger.debug()` with `emit.debug()` after importing `emit` from `craft_cli`.

Reply:
This might be a bit trickier to do. I can switch to the `craft_cli` `emitter` but this will likely lover coverage since some of the tests are checking log messages. Unless there is a way to capture messages emitted by `craft_cli`

Reply:
You can use the `emitter` test fixture defined by craft-cli to test the output.